blackbird_python/blackbird/tests/test_listener.py

```diff
@@ -346,6 +347,313 @@ def test_target_arg(self, parse_input):
             parse_input("name testname\nversion 1.0\ntarget example (6)")
 
 
+class TestParsingInclude:
+    """Tests for the `include` statement"""
+
+    @pytest.mark.skipif(sys.version_info < (3, 6), reason="tmpdir fixture requires Python >=3.6")
+    def test_include_program(self, parse_input, tmpdir):
+        """Test including a non-templated program"""
+        program = textwrap.dedent(
+            """
+            name CustomOperation
+            version 0.0
+            float alpha = 0.3423
+            Coherent(alpha, sqrt(pi)) | 0
+            MeasureFock() | 0
+            """
+        )
+
+        filename = tmpdir.join("test.xbb")
+
+        with open(filename, "w") as f:
+            f.write(program)
+
+        test_include = textwrap.dedent(
+            """
+            name test_include
+            version 0.0
+            include "{}"
+            CustomOperation | 1
+            """
+        ).format(filename)
+
+        bb = parse_input(test_include, cwd=tmpdir)
+
+        expected = [
+            {"op": "Coherent", "args": [0.3423, np.sqrt(np.pi)], "kwargs": {}, "modes": [1]},
+            {"op": "MeasureFock", "args": [], "kwargs": {}, "modes": [1]},
+        ]
+
+        assert bb.operations == expected
+
+    @pytest.mark.skipif(sys.version_info < (3, 6), reason="tmpdir fixture requires Python >=3.6")
+    def test_include_template(self, parse_input, tmpdir):
+        """Test including a templated program"""
+        template = textwrap.dedent(
+            """
+            name CustomOperation
+            version 0.0
+            BSgate({theta}, pi/2) | [0, 1]
+            Rgate({phi}) | 0
+            """
+        )
+
+        filename = tmpdir.join("test.xbb")
+
+        with open(filename, "w") as f:
+            f.write(template)
+
+        test_include = textwrap.dedent(
+            """
+            name test_include
+            version 0.0
+            include "{}"
```

Comment:
Go a bit confused here at first, since curly braces are used for templating, but also for string formatting

Reply:
Yep, exactly 😄